ADAL/src/telemetry/ADTelemetryDefaultEvent.m

```diff
@@ -125,6 +125,19 @@ - (NSString*)getStringFromDate:(NSDate*)date
     return [s_dateFormatter stringFromDate:date];
 }
 
+- (void)sanitizePropertyNames
+{
+    NSMutableDictionary* propertyMapCopy = [_propertyMap copy];
+    [_propertyMap removeAllObjects];
+    
+    for (NSString* propertyName in propertyMapCopy)
+    {
+        // Some telemetry collectors only allow alphabetic characters, underscores, and dots in the property names
+        [_propertyMap setObject:[propertyMapCopy objectForKey:propertyName]
+                         forKey:[propertyName stringByReplacingOccurrencesOfString:@"-" withString:@"_"]];
```

Comment:
If we're going to be not allowing hyphens in property names/keys then we should be changing the key values themselves.